zuul/model.py

```diff
@@ -1431,8 +1431,19 @@ def matches(self, change):
                 return False
 
         if self.statuses:
-            if change.status not in self.statuses:
-                return False
+            # handle a change where status is list of statuses rather
+            # than a single string
+            if not isinstance(change.status, list):
```

Comment:
nit: always found the best thing to do when testing lists is make it what you want.

statuses = change.status if isinstance(change.status, list) else [change.status]